base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -902,6 +906,27 @@ where
             },
             UTXOSelectionStrategy::Largest => uo.into_iter().rev().collect(),
         };
+        let trimmed = uo.len();
+        trace!(target: LOG_TARGET, "We found {} UTXOs to select from", trimmed);
+
+        // Now lets try to filter the UTXOs we cant yet spend.
+        let tip_height: u64 = self
+            .base_node_service
+            .get_connected_base_node_state()
+            .await?
+            .height_of_longest_chain();
+        uo = uo
```

Comment:
nit: This is inefficient with a large number of utxos. It will be better to combine this filter with the next `iter`